### PR TITLE
[suggestion] macOS Riot icon

### PR DESCRIPTION
**Pull request by alicerunsonfedora**
In the Riot chat, we discussed bringing a new macOS-specific icon for Riot to make it more in-line with other Mac apps. This is my request to add that icon in (but I may have added the code in the wrong ```package.json``` file) using this: https://www.github.com/electron-userland/electron-builder/wiki/Options#user-content-MacOptions-icon

Either way, there's a new icon file in the img folder of the electron_app folder (if that makes sense).

EDIT: Fixed tabbing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -168,6 +168,9 @@
         "StartupWMClass": "riot"
       }
     },
+      "mac": {
```
Comment (t3chguy) [inline package.json]: please fix your indentation

**Comment by t3chguy** [comment]
You also must follow the contribution doc, specifically Sign-Off:
https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off

**Comment by ubunturox104** [comment]
Signed-off-by: Marquis Kurt <ubunturox104@gmail.com>

**Comment by t3chguy** [comment]
riot

^^^ PNG for easier viewing for Windows users ^^^

structurally PR looks good and should work, now its down to whether the logo is wanted/liked :)

**Comment by ara4n** [comment]
@ubunturox104 thanks for this! sorry for the delay in response - we've been having a slightly nightmarish week.  The icon is an improvement on what we have currently, although I'm not entirely sure that it's completely on brand.  I'm going to ask the folks who originally designed the logo if they have a macOS variant which we can add into the PR here.

**Comment by ubunturox104** [comment]
@ara4n No worries. I tried to base it off of the Riot website in terms of the gradient and pulled one of the Riot vector images from the older website:
img_0354

Primarily, I made this to make it more in line with other macOS icons, especially with Apple’s Human Interface Guidelines. I understand if it may need to be changed a bit to better suit the brand.

**Comment by rodneyrod** [comment]
Any news on this?

**Comment by ara4n** [comment]
it's tricky. i like the logo, and it's better than the current one. but for consistency with the other platforms (iOS & Android) I think we should be using the purple-on-concentric-green variant: https://github.com/vector-im/logos/blob/master/riot/riot-im-logo-1.svg

If this was put inside a white/grey plinth I'd merge it like a shot.  Sorry for being picky :(

**Comment by ara4n** [comment]
(and sorry that this has dragged on for months)

**Comment by lukebarnard1** [comment]
>If this was put inside a white/grey plinth I'd merge it like a shot

@alicerunsonfedora, would you be able to have another go, using https://github.com/vector-im/logos/blob/master/riot/riot-im-logo-1.svg ?

**Comment by dbkr** [comment]
Please also not that the raster versions of these icons are generated from vector versions. We'd need a vector version to be able to merge this I'd think, otherwise we'd be stuck having to redraw it if we needed any changes made.

**Comment by alicerunsonfedora** [comment]
@lukebarnard1 I'll see what I can do to update the icon. Anny suggestions for the background color of the icon?

**Comment by lukebarnard1** [comment]
@alicerunsonfedora I'd have to defer to @ara4n on that one. Riot iOS and Android both appear to use a white background.

**Comment by alicerunsonfedora** [comment]
Any thoughts on this redesign?
[riot-mac-icon.svg.zip](https://github.com/vector-im/riot-web/files/2389148/riot-mac-icon.svg.zip)

**Comment by HansJK** [comment]
Like the purple one. The "RIOT.IM" text takes away the clean design. But since Riot is being rebranded, there might be new possibilities for a new icon. I think it's more important that the icon looks clean and nice even if it's not 100% with the brand. Personally I made this one for personal use. Never liket the standard icon. The blue color is from the sidebar from the new Riot design. Hope to see a new clean icon with the 1.0 realise.
screenshot 2019-02-13 at 01 05 22

**Comment by alicerunsonfedora** [comment]
@HansJK I like that design! I'm going to try a 'version three' by combining your design and the brand colors to see if I can come up with something.

**Comment by HansJK** [comment]
@alicerunsonfedora Looking forward to seeing what you can come up with. The current one is just not professional enough.
screenshot 2019-02-14 at 22 51 03

**Comment by alicerunsonfedora** [comment]
I have a few different ideas, though I feel they need some refinements.

aaaaaa

**Comment by HansJK** [comment]
The logo lines should maybe be a little thicker, so they don't disappear in small dock. Also the double line at the bottom should be a single line instead. I think the logo need to be a little further down, because it looks like it's further to the top than the bottom. I kinda liked when the logo was a little tilted like my version, but still looks good. Overall a huge improvements to the old one you made.

**Comment by alicerunsonfedora** [comment]
@HansJK Yeah, it does kinda look off, though if you take Apple's grid, it's perfectly aligned. Weird! As for the "double line", I think that's just the result of how the SVG for the logo was made...

Whenever I can, I'll try to think of some improved designs. Anyone of the three you think particularly looks good?

**Comment by HansJK** [comment]
@alicerunsonfedora The first one

**Comment by HansJK** [comment]
This is my first try after the redesign. I used the two blue colours from the current Riot logo, (Blue to lighter blue) in this gradient, so it works with Riot's branding. 
riot-mac-icon

**Comment by alicerunsonfedora** [comment]
I personally would up the saturation on the blue a little bit and make the R itself match with the outer metal. I'll get a render of it later.

**Comment by HansJK** [comment]
Agree that the saturation should be turned ip, it would look more modern. But the problem is that Riot's branding use somewhat dated colours. This colors where taken from the logo itself. I don't think they would like an icon that dosen't use their branding colors. 

Don't think the (R) would work well with a gray gradient. But I can try. Btw, I used your border.

**Comment by HansJK** [comment]
Made some new versions of the icon
riot-mac-icon

**Comment by alicerunsonfedora** [comment]
I like number 6 in particular, though I'd change the logo itself so it acted more like number 3 (ish) and follow the flow of the outer gradient.

**Comment by HansJK** [comment]
Added gradient on the on the Riot logo in number 7. I think I like it better as it is in number 6. Not sure if this is what you meant or not.
riot-mac-icon

**Comment by alicerunsonfedora** [comment]
Yeah, that's close enough. I was about to make one that resembled it

**Comment by HansJK** [comment]
Here I have SVG and ICNS of number 6 and 7.
[riot_icon.zip](https://github.com/vector-im/riot-web/files/2921697/riot_icon.zip)

**Comment by alicerunsonfedora** [comment]
Awesome, thanks! I'll make a few minor adjustments to logo 7 and will get to work on getting the PR rolling again, granted that the 'highlight-js' problem gets resolved (hence why Travis has been failing constantly).

**Comment by alicerunsonfedora** [comment]
Also, @t3chguy Is there a way to do a sign-off on behalf of more than one person? I'd like to give credit to @HansJK as well.

**Comment by aaronraimist** [comment]
You can do co-authored-by:

Example https://github.com/matrix-org/matrix-doc/commit/6e8739c989172c351a6ead66f073b413f6340326

**Comment by HansJK** [comment]
@alicerunsonfedora What kind of adjustments are you thinking about?

**Comment by alicerunsonfedora** [comment]
Mostly drop shadows.

**Comment by alicerunsonfedora** [comment]
riot icon 7 improvement sheet

**Comment by HansJK** [comment]
If it is a problem with the colors in Gravitdesigner, i can just modify the shadow in Illustrator.

**Comment by alicerunsonfedora** [comment]
Attached are the redesigns I just did as SVG. On second glance, I'm thinking Affinity Photo skewed as the SVGs appear normal now... (I use P3 in Affinity Photo)

[r7_improvements.zip](https://github.com/vector-im/riot-web/files/2922498/r7_improvements.zip)

**Comment by HansJK** [comment]
Just so we are on the same page, is the r1 you included this one? I prefer this one or the one where the R is tilted. Also is the colors fine now, Affinity didn't accidentally change anything? :)
screenshot 2019-03-02 at 23 52 49

**Comment by alicerunsonfedora** [comment]
Both R1 and R2 use the straight icon. I don't really think the diagonal really works here as those are usually reserved for the rectangular shapes like the Pages icon. Though it is ultimately up to the community whether they want the straight one or not.

Also, GravitDesigner seemed to handle the colors just fine, but when I imported the PNG versions of my exported SVGs into Affinity Photo to make the sheet from earlier, that's when the colors were skewed. It happened because I use the P3 color spectrum/gamut thingy for it (to match my MacBook's display) and thus "richened" the colors.

**Comment by HansJK** [comment]
Yeah I think the strait one has the biggest chance of making it. Really happy with how it turned out with he fill.

**Comment by alicerunsonfedora** [comment]
I have no clue if that worked correctly as I don't have @HansJK 's GitHub email address, but hopefully, that should suffice. The commit includes both the ICNS file and the SVG version for rendering.

**Comment by alicerunsonfedora** [comment]
Here's how it looks on the Dock:
screen shot 2019-03-03 at 13 43 27

**Comment by HansJK** [comment]
Looks amazing!

**Comment by alicerunsonfedora** [comment]
I just discovered that the sizing of the icon was off when I was working on a new Mac icon for Hyperspace. I just resized it and made the changes here. Again, looks like co-authoring didn't work, but it hopefully should still give attributes as necessary.

**Comment by nadonomy** [comment]
Hey @alicerunsonfedora @HansJK,

Thanks for putting the time and effort into this. Apologies for not being a part of the previous discussions. From my side, there's a few blockers that stand in the way of merging:

1. Internally we use Sketch for almost all of our vector based work, along with treating some assets with Adobe Creative Cloud tools for print. While exporting for web/desktop can be more forgiving, we can't guarantee other tools will work well in all contexts (colour profiles and print generally being the most fussy). As a small team, out of pragmatism we need to ensure community contributions are authored using a pipeline we work with.
2. I'm concerned about using perspective and texture to style the icon to look like a native macOS app. I don't think it's healthy signalling for the Electron app.
3. Drastically varying the icon per platform dilutes brand equity and is confusing for a lot of users, along with setting a precedent that we adhere individually to each platform we package for. We could consider using the same mark but varying how we crop and treat the design on different platforms, though.
4. The mark itself has been modified, for example the stem lacks a rounded cap.

Apologies that these points weren't raised before you spent time on it— we've had the electron icon on our internal agenda for a while and we weren't anticipating external activity on it.

If you're interested in working through the above, please [message me on Matrix](https://matrix.to/#/@nadonomy:matrix.org) and I can share some internal assets with you— including some of the existing icon source files, and some brand guidelines that have helped inform the redesign to date.

Cheers,
Nad

**Comment by jryans** [comment]
Given Nad's comment above, I'll close this PR for now. Anyone interested in this topic should reach out to Nad so we can work out the right path forward. Thanks again for everyone's work here! 😄

**Pull request closed without merging**